homeassistant/components/pushbullet/sensor.py

```diff
@@ -128,11 +127,24 @@ async def async_setup_entry(
 
     pushbullet: PushBullet = hass.data[DOMAIN][entry.entry_id]
     pb_provider = PushBulletNotificationProvider(hass, pushbullet)
+
+    def listener_start(event: Event) -> None:
+        """Start the listener thread."""
+        _LOGGER.debug("Starting listener for pushbullet")
+        pb_provider.start()
+
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, listener_start)
+    entry.async_on_unload(pb_provider.close)
```

Comment:
Is `pb_provider.close` async safe? Otherwise we need to schedule it on the executor from `async_unload_entry` in the init module.

Reply:
To achieve I moved the `PushBulletNotificationProvider` class to `api.py` and stored this in hass.data. This way I can close the thread from `async_unload_entry` and move the initialization and start of the class to `async_entry_setup`.